src/azure/cli/parser.py

```diff
@@ -62,11 +65,24 @@ def load_command_table(self, command_table):
                                                   help_file=metadata.help)
 
             argument_validators = []
+            argument_groups = {}
             for arg in metadata.arguments.values():
                 if arg.validator:
                     argument_validators.append(arg.validator)
-                param = command_parser.add_argument(
-                    *arg.options_list, **arg.options)
+                if arg.arg_group:
+                    try:
+                        group = argument_groups[arg.arg_group]
+                    except KeyError:
+                        # group not found so create
+                        name_comps = re.findall('[A-Z][^A-Z]*', arg.arg_group)
+                        group_name = '{} Arguments'.format(' '.join(name_comps))
```

Comment:
Technically yes, but I would like the groups to be registered to parameters by a key, not a title. So ideally in the register_parameter call, arg_group would be something like "storage" and not "Storage Account". I'm working on that.

---
In reply to: [74973634](https://github.com/Azure/azure-cli/pull/706#discussion_r74973634) [](ancestors = 74973634)